tools/schemapi/vega_expr.py

```diff
@@ -0,0 +1,964 @@
+        msg = (
+            f"Expressions download failed: {fp!s}.\n\n"
+            f"Try manually accessing resource: {url!r}"
+        )
+        raise FileNotFoundError(msg)
+    else:
+        return fp
+
+
+def strip_include_tag(s: str, /) -> str:
+    """
+    Removes `liquid`_ templating markup.
+
+    .. _liquid:
+        https://shopify.github.io/liquid/
+    """
+    return LIQUID_INCLUDE.sub(r"", s)
+
+
+def expand_urls(url: str, /) -> str:
+    if url.startswith("#"):
+        url = f"{EXPRESSIONS_DOCS_URL}{url}"
+    else:
+        url = url.replace(r"../", VEGA_DOCS_URL)
+    return url
+
+
+class RSTRenderer(_RSTRenderer):
+    def __init__(self) -> None:
+        super().__init__()
+
+    def link(self, token: Token, state: BlockState) -> str:
+        """
+        Store link url, for appending at the end of doc.
+
+        TODO
+        ----
+        - Parameterize `"#"`, `"../"` expansion during init
+        """
+        attrs = token["attrs"]
+        url = expand_urls(attrs["url"])
+        text = self.render_children(token, state)
+        text = text.replace("`", "")
+        inline = f"`{text}`_"
+        state.env["ref_links"][text] = {"url": url}
+        return inline
+
+    def text(self, token: Token, state: BlockState) -> str:
+        text = super().text(token, state)
+        return strip_include_tag(text)
+
+    def _with_links(self, s: str, links: dict[str, Any] | Any, /) -> str:
+        it = chain.from_iterable(
+            (f".. _{ref_name}:", f"    {attrs['url']}")
+            for ref_name, attrs in links.items()
+        )
+        return "\n".join(chain([s], it))
+
+    def __call__(self, tokens: Iterable[Token], state: BlockState) -> str:
+        result = super().__call__(tokens, state)
+        if links := state.env.get("ref_links", {}):
+            return self._with_links(result, links)
+        else:
+            return result
+
+
+parser: RSTParse = RSTParse(RSTRenderer())
+text_wrap = _TextWrapper(
+    width=100,
+    break_long_words=False,
+    break_on_hyphens=False,
+    initial_indent=8 * " ",
+    subsequent_indent=8 * " ",
+)
+
+
+def _doc_fmt(doc: str, /) -> str:
+    """
+    FIXME: Currently doing too many things.
+
+    Primarily using to exclude summary line + references from ``textwrap``.
+    """
+    sentences: deque[str] = deque(SENTENCE_BREAK.split(doc))
+    if len(sentences) > 1:
+        references: str = ""
+        summary = f"{sentences.popleft()}.\n"
+        last_line = sentences.pop().strip()
+        sentences = deque(f"{s}. " for s in sentences)
+        if "\n\n.. _" in last_line:
+            last_line, references = last_line.split("\n\n", maxsplit=1)
+        sentences.append(last_line)
+        sentences = deque(text_wrap.wrap("".join(sentences)))
+        sentences.appendleft(summary)
+        if references:
+            sentences.extend(("", indent(references, 8 * " ")))
+        return "\n".join(sentences)
+    else:
+        return sentences.pop().strip()
```

Comment:
### Issues
- Very tightly coupled to how `RSTRenderer` appends links
- Don't think `re.split` is that intuitive, despite working for this specfic case
- Would prefer something more structured